packages/plugin-ext/src/hosted/node/hosted-plugin-localization-service.ts

```diff
@@ -140,12 +194,58 @@ export class HostedPluginLocalizationService implements BackendApplicationContri
     }
 
     protected async getLocalizationCacheDir(): Promise<string> {
-        const configDir = URI.parse(await this.envVariables.getConfigDirUri()).fsPath;
+        const configDir = new URI(await this.envVariables.getConfigDirUri()).path.fsPath();
         const cacheDir = path.join(configDir, 'localization-cache');
         return cacheDir;
     }
 }
 
+// New plugin localization logic using vscode.l10n
+
+function getL10nTranslation(translation: Translation): UnprocessedL10nBundle | undefined {
+    // 'bundle' is a special key that contains all translations for the l10n vscode API
+    // If that doesn't exist, we can assume that the language pack is using the old vscode-nls API
+    return translation.contents.bundle;
+}
+
+async function loadPluginBundles(l10nUri: URI): Promise<Record<string, LanguagePackBundle> | undefined> {
+    try {
+        const directory = l10nUri.path.fsPath();
+        const files = await fs.readdir(directory);
+        const result: Record<string, LanguagePackBundle> = {};
+        await Promise.all(files.map(async fileName => {
+            const match = fileName.match(/^bundle\.l10n\.([\w\-]+)\.json$/);
+            if (match) {
+                const locale = match[1];
+                const contents = await fs.readJSON(path.join(directory, fileName));
+                result[locale] = {
+                    contents,
+                    uri: l10nUri.resolve(fileName).toString()
+                };
+            }
+        }));
+        return result;
+    } catch {
+        // The directory either doesn't exist or its contents cannot be parsed
```

Comment:
Logging this seems like a good idea, though. Think of some developer trying to do i18n and "it's just not working".

Reply:
Not sure if this should be an error or a warning. If I understand correctly, there are legitimate cases that will also end up inside this catch block?